File: Blog.Services/CommentDTO.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Blog.Services
{
    public class CommentDTO
    {
        public int CommentId { get; set; }

        [Required]
        public string Content { get; set; }

        public DateTime CreateDate { get; set; }

        public string UserName { get; set; }

        public int RecordId { get; set; }

        public RecordDTO Record { get; set; }
    }
}

File: Blog.ReactUI/Src/Types/CommentDTO.ts
import RecordDTO from "types/RecordDTO";
import UserDTO from "types/UserDTO";

export default class CommentDTO {
  public CommentId: number;

  public Content: string;

  public CreateDate: Date;

  public RecordId: number;

  public Record: RecordDTO;

  public UserId: number;

  public User: UserDTO;
}
